[docs] Update "About notification types" page (#36269)
expo/expo: docs/pages/push-notifications/what-you-need-to-know.mdx

description: Learn about notification types and their behavior before you get started.
---

> Notifications are alerts that inform users of new information or events, even when the app isn't actively in use. Notifications have a large surface area and differences across platforms can make implementing notifications intimidating.

Reviewer: ```suggestion
Notifications are alerts that inform users of new information or events, even when the app isn't actively in use. They have a large surface area and differences across platforms can make implementing notifications intimidating.
```

> Whether you are just starting with notifications or have existing knowledge, this document explains the different types of notification and their behaviors.

Reviewer: ```suggestion
Whether you are starting with notifications or have existing knowledge, this document explains the different types of notification and their behaviors.
```

I'd suggest avoiding using that. I don't think it has any significance.

Expo's notification support builds on top of the native functionality provided by Android and iOS. The same concepts and behaviors from native platforms apply to Expo apps. If you are unsure about a specific notification feature, see each platform's [official documentation](#external-references).

> ## Remote and Local Notifications

Reviewer: All the section titles follow the sentence case. Please revert them in orginial state. For more info, check: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings.

Author: Then should it not be "Remote and local notifications" ?  What's the rule about **Local** being capitalized but **notifications** not?  This seems inconsistent to me.  And could it not be argued that **Notifications** is the name of a product (or at least a feature)?

I defer to your call.  Just pointing out a confusion as I see it.

1. **Push Notifications**: (also known as "remote notifications") Notifications that are sent to a user's device from a remote server.
2. **Local Notifications**: (also known as "in-app notifications") Notifications that are created and displayed from within the app. Since many of the APIs that create these notifications will create them at a particular time, these may also sometimes be called "scheduled notifications".

`expo-notifications` supports both push and local notifications. You must use a [development build](/develop/development-builds/introduction/) to use push notifications since the capability is not built into Expo Go.

> See [in-app notifications](/versions/latest/sdk/notifications/#present-a-local-in-app-notification-to-the-user) for how to create and display a local notification.  The rest of this guide focuses on push notifications.

Reviewer: ```suggestion
See [in-app notifications](/versions/latest/sdk/notifications/#present-a-local-in-app-notification-to-the-user) on how to create and display a local notification. The rest of this guide focuses on push notifications.
```

## Push Notification Delivery

When a push notification arrives to your app, its behavior depends on the app's state and the type of notification. Let's clarify the terminology:

### Application States

> - **Foreground**: The app is actively running in the foreground; its interface is currently being displayed on the screen.
> - **Background**: The app is running in the background, "minimized"; its interface is not currently being displayed on the screen.

Reviewer: ```suggestion
- **Foreground**: The app is actively running in the foreground. Its interface is currently being displayed on the screen.
- **Background**: The app is running in the background, "minimized". Its interface is not currently being displayed on the screen.
```

In the table above, whenever `NotificationResponseReceivedListener` is triggered, the same would apply to the `useLastNotificationResponse` hook.

> > **info** When the app is not running or killed and is restarted by tapping on a notification, the `NotificationResponseReceivedListener` may or may not be triggered. To reliably capture the response, we recommend using `useLastNotificationResponse` or `getLastNotificationResponseAsync`. Expo expects to improve this behavior in a future release.

Reviewer: ```suggestion
> **info** When the app is not running or killed and is restarted by tapping on a notification, the `NotificationResponseReceivedListener` may or may not be triggered. To reliably capture the response, we recommend using `useLastNotificationResponse` or `getLastNotificationResponseAsync`. We plan to improve this behavior in a future release.
```

The original language is fine since we reserve "we" to convey a message/info from Expo team.

The rule of thumb is to prefer a regular Notification Message if you don't require running JavaScript in the background.

> \* The exception is when you specify `title` or `message` inside of [`data`](https://github.com/expo/expo-server-sdk-node/blob/88ed8fbfc1345f13911d8a8d30a5d8009aaa8892/src/ExpoClient.ts#L349). In that case, `expo-notifications` package automatically presents the headless notification on Android, but not on iOS. Expo plans to make this behavior more consistent across platforms in a future release.

Reviewer: ```suggestion
\* The exception is when you specify `title` or `message` inside of [`data`](https://github.com/expo/expo-server-sdk-node/blob/88ed8fbfc1345f13911d8a8d30a5d8009aaa8892/src/ExpoClient.ts#L349). In that case, `expo-notifications` package automatically presents the headless notification on Android, but not on iOS. We plan to make this behavior more consistent across platforms in a future release.
```

Same as previous
